### PR TITLE
Implement #111 - Add methods for Forestry bee breeding data

### PR DESCRIPTION
**Pull request by dmarcuse**
Here's an implementation for #111:

- Uses Maven for Forestry so the API is available
- Bumps JEI version to prevent runtime conflicts (Forestry requires a newer version of JEI)
- Adds `getSpeciesRoots`, `getSpeciesList`, and `getMutationsList` to the portable analyzer when used in a manipulator (secret species/mutations are not included of course)
- Adds `LuaList`, simple wrapper type for returning lists

Screenshots:

- `getSpeciesRoots`
image

- `getSpeciesList`
image

- `getMutationsList`
image
image

### CODE REVIEW

**Comment by SquidDev** [comment]
OK, I'm going to need to look into why CI is failing.

File: src/main/java/org/squiddev/plethora/integration/forestry/MethodsBeeHousing.java [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+import forestry.api.apiculture.IAlleleBeeSpecies;
 import forestry.api.apiculture.IBeeHousing;
+import forestry.api.apiculture.IBeeRoot;
 import forestry.api.core.IErrorState;
+import forestry.api.genetics.*;
 import forestry.core.config.Constants;
 import net.minecraft.item.ItemStack;
 import org.squiddev.plethora.api.method.BasicObjectMethod;
 import org.squiddev.plethora.api.method.IContext;
+import org.squiddev.plethora.utils.DebugLogger;
```
Comment (SquidDev) [inline src/main/java/org/squiddev/plethora/integration/forestry/MethodsBeeHousing.java]: Can you remove this line? I don't think the import is used anywhere.

**Review by SquidDev — CHANGES_REQUESTED** [review]

**Comment by dmarcuse** [comment]
The latest commit doesn't seem to work with the analyzer block, and triggers the following NPE when using `listModules` or `filterModules` on it (portable analyzer + manipulator still works fine). Could you take a look @SquidDev?

```
[21:38:26] [Coroutine-3/ERROR]: Unexpected error calling filterModules
java.lang.NullPointerException: null
	at org.squiddev.plethora.integration.MethodsModules.filterModules(MethodsModules.java:47) ~[MethodsModules.class:?]
	at org.squiddev.plethora.integration.MethodsModules$filterModules.apply(Unknown Source) ~[?:?]
	at org.squiddev.plethora.api.method.BasicObjectMethod.apply(BasicObjectMethod.java:42) ~[BasicObjectMethod.class:?]
	at org.squiddev.plethora.core.MethodWrapper.doCallMethod(MethodWrapper.java:86) [MethodWrapper.class:?]
	at org.squiddev.plethora.core.MethodWrapperPeripheral.callMethod(MethodWrapperPeripheral.java:78) [MethodWrapperPeripheral.class:?]
```

File: build.gradle [hunk 2]
```diff
@@ -80,6 +80,9 @@ repositories {
 
 	// CBMP
 	maven { url "http://chickenbones.net/maven" }
+
+	// Forestry
+	maven { url "http://maven.ic2.player.to/" }
```
Comment (SquidDev) [inline build.gradle]: Can we merge this with line 62? Sorry, should have seen this earlier.

File: src/main/java/org/squiddev/plethora/integration/forestry/IntegrationForestry.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			MinecraftForge.EVENT_BUS.register(new IntegrationForestry());
+		}
+	}
+
+	private static BasicModuleHandler _analyzerCap;
+
+	@Optional.Method(modid = Constants.MOD_ID)
+	private static BasicModuleHandler getAnalyzerCap() {
+		if (_analyzerCap == null) {
+			_analyzerCap = new BasicModuleHandler(new ResourceLocation(analyzerMod), ModuleCore.getItems().portableAlyzer);
```
Comment (SquidDev) [inline src/main/java/org/squiddev/plethora/integration/forestry/IntegrationForestry.java]: No underscore.

File: src/main/java/org/squiddev/plethora/integration/forestry/MethodsErrorLogicSource.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import org.squiddev.plethora.utils.LuaList;
+
+public class MethodsErrorLogicSource {
+	@BasicObjectMethod.Inject(
+			value = IErrorLogicSource.class, modId = Constants.MOD_ID, worldThread = true,
+			doc = "function():table -- Get any errors preventing operation"
+	)
+	public static Object[] getErrors(IContext<IErrorLogicSource> context, Object[] arg) {
+		return new Object[] {
+				context.getTarget().getErrorLogic().getErrorStates().stream()
```
Comment (SquidDev) [inline src/main/java/org/squiddev/plethora/integration/forestry/MethodsErrorLogicSource.java:17]: I'm so going to convert this into a loop at some point :p. I'll live with it for now though.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Just a couple of minor nitpicky things. Sorry.

**Review by SquidDev — APPROVED** [review]

File: src/main/java/org/squiddev/plethora/integration/forestry/MethodsErrorLogicSource.java (hunk 4 above)
Comment (dmarcuse) [reply src/main/java/org/squiddev/plethora/integration/forestry/MethodsErrorLogicSource.java:17]: Yeah, this one may have been a bit gratuitous :stuck_out_tongue:

**Comment by SquidDev** [comment]
Thank you very much!

**Pull request merged**